code/modules/mob/dead/observer/orbit.dm

```diff
@@ -18,94 +34,132 @@
 
 /datum/orbit_menu/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
 	. = ..()
+
 	if(.)
 		return
+
 	switch(action)
 		if ("orbit")
 			var/ref = params["ref"]
-			var/atom/movable/poi = (locate(ref) in GLOB.mob_list) || (locate(ref) in GLOB.poi_list)
-			if (poi == null)
-				. = TRUE
-				return
-			owner.ManualFollow(poi)
-			owner.reset_perspective(null)
+			var/auto_observe = params["autoObs"]
+			var/atom/movable/poi = pois[ref]
+			if (!poi)
+				to_chat(usr, span_notice("That point of interest is no longer valid."))
+				return TRUE
+
+			var/mob/dead/observer/user = usr
+			user.ManualFollow(poi)
+			user.reset_perspective(null)
 			if (auto_observe)
-				owner.do_observe(poi)
-			. = TRUE
+				user.do_observe(poi)
+			return TRUE
 		if ("refresh")
-			update_static_data(owner, ui)
-			. = TRUE
-		if ("toggle_observe")
-			auto_observe = !auto_observe
-			if (auto_observe && owner.orbit_target)
-				owner.do_observe(owner.orbit_target)
-			else
-				owner.reset_perspective(null)
-
-/datum/orbit_menu/ui_data(mob/user)
-	var/list/data = list()
-	data["auto_observe"] = auto_observe
-	return data
+			update_static_data(usr, ui)
+			return TRUE
 
 /datum/orbit_menu/ui_static_data(mob/user)
 	var/list/data = list()
 
-	var/list/alive = list()
-	var/list/antagonists = list()
-	var/list/dead = list()
-	var/list/ghosts = list()
-	var/list/misc = list()
-	var/list/npcs = list()
+	data["alive"] = alive
+	data["antagonists"] = antagonists
+	data["dead"] = dead
+	data["ghosts"] = ghosts
+	data["misc"] = misc
+	data["npcs"] = npcs
+
+	return data
+
+/datum/orbit_menu/ui_assets()
+	. = ..() || list()
+	. += get_asset_datum(/datum/asset/simple/orbit)
```

Comment:
```suggestion
	return list(
		get_asset_datum(/datum/asset/simple/orbit),
	)
```

I have no clue why I did it like this but I know better now.

Reply:
Changed.